pkg/morph/client/notary.go

```diff
@@ -93,17 +102,27 @@ func (c *Client) EnableNotarySupport(opts ...NotaryOption) error {
 		alphabetSource: cfg.alphabetSource,
 	}
 
-	return nil
-}
+	// update client cache
+	for _, cached := range c.clients {
+		cached.notary = c.sharedNotary
+	}
```

Comment:
This loop sets `sharedNotary` structure for already created clients. I suppose new clients will miss it?

Reply:
New clients has same setter of the field in `createForAddress` method. Since `EnableNotarySupport` can be called at any time, I added this loop to ensure all clients work in the same mode. I did it as an alternative of pre-clearing the cache before turning on the notary.

Reply:
I see, all new clients created by `createForAddress` will already have shared notary, missed that
```go
	c := &Client{
		singleClient: &singleClient{
			notary:       x.sharedNotary,
		},
	}
```